src/main/java/online/shop/service/ProductService.java

```diff
@@ -0,0 +1,19 @@
+package online.shop.service;
+
+import java.util.List;
+import java.util.Optional;
+import online.shop.model.Product;
+import online.shop.service.lib.Service;
+
+@Service
+public interface ProductService {
+    Product create(Product product);
+
+    Optional<Product> getById(Long productId);
```

Comment:
I think in Service we need to return Product, not Optional<Product>. We should return Optional<Product> only in Dao.